spec/workers/data_update_worker_spec.rb

```diff
@@ -55,4 +55,8 @@
       expect(DatadogStatsClient).to have_received(:increment).once.with(*expected_args)
     end
   end
+
+  xit "saves an error to the database" do
```

Comment:
**I'm struggling a bit with this test and would love some insights:**

I want to test this code:
```
  def run_script(script)
    ....
  rescue StandardError => e
    script.mark_as_failed!
    log_status(script)
    script.save_error!(e) # THIS LINE NEEDS TO BE TESTED

    Honeybadger.notify(e, context: { script_id: script.id })
  end
```

I want the test to throw an error, and let the rescue block catch it and run the code in the block. Thereafter I want to pull up the last data update script and see that the error was saved. However, the part that I'm struggling with is how to make sure the code in the rescue block is executed.

Reply:
You can stub `.mark_as_finished!`  to throw an error and then check that the `error` column has been populated. I think it should work